lib/src/disposable/disposable.dart

```diff
@@ -189,6 +190,19 @@ class Disposable implements _Disposable, DisposableManagerV2 {
   /// and will become `false` once the [didDispose] future completes.
   bool get isDisposing => _isDisposing;
 
+  @mustCallSuper
+  @override
+  Future<T> awaitBeforeDispose<T>(Future<T> future) {
```

Comment:
The difference is that the list is tidied up in this case, using `manageDisposer` would accumulate incomplete futures forever.